drivers/linux/chipsec_km.c

```diff
@@ -560,13 +560,24 @@ int __weak phys_mem_access_prot_allowed(struct file *file,
         return 1;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,1,12) && defined(ARCH_HAS_VALID_PHYS_ADDR_RANGE) && !defined(__HAVE_ARCH_PAX_OPEN_USERLAND)
+#if defined(__HAVE_ARCH_PAX_OPEN_USERLAND)
+#if LINUX_VERSION_CODE > KERNEL_VERSION(4,14,0) && defined(ARCH_HAS_VALID_PHYS_ADDR_RANGE)
 int valid_mmap_phys_addr_range(unsigned long pfn, size_t size)
 {
 	return 1;
 }
 #endif
 
+#else
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,1,12) && defined(ARCH_HAS_VALID_PHYS_ADDR_RANGE)
```

Comment:
Is there any specific reason why this condition is separate from from the one from L564?

Reply:
the idea was: If the kernel is PaX/Grsec > v4.14, it'll use the "return 1" hack since grsec seems didn't impl valid_mmap_phys_addr_range() by their own. Otherwise, it'll just keep what the CHIPSEC does.